dataset/variable_reduction.cpp

```diff
@@ -62,8 +66,9 @@ Variable nanmean(const Variable &var, const Dim dim, const Masks &masks) {
   using variable::isfinite;
   if (const auto mask_union = irreducible_mask(masks, dim);
       mask_union.is_valid()) {
-    const auto count = sum(masked_to_zero(isfinite(var), mask_union), dim);
-    return nanmean_impl(masked_to_zero(var, mask_union), dim, count);
+    const auto finite = isfinite(var);
```

Comment:
For consistency with numpy, this should be `!isnan`. E.g. `np.nanmean([1, 2, np.inf])` returns `np.inf`.

Reply:
So does scipp? Not sure I understand your point.

```python
sc.nanmean(sc.array(dims=['x'], values=[1, 2, np.inf]))
```